qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -51,6 +69,18 @@ def run(self, dag):
         Raises:
             TranspilerError: when the 1-qubit rotation gates are not found
         """
+        var_z_gate = None
+        z_var_gates = [gate for gate in dag.properties()['operations'].keys()
```

Comment:
```suggestion
        z_var_gates = [gate for gate in dag.count_ops().keys()
```
Might save from computing some of the unneeded properties.

Reply:
thanks